interface/src/app/volcano-plot/volcano-plot.js

```diff
@@ -4,50 +4,134 @@
  * via plotData.
  */
 angular.module('adage.volcano-plot', [
+  'ngResource',
   'ngVega',
   'greenelab.stats',
+  'adage.utils',
   'adage.volcano-plot-vgspec'
 ])
 
-.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec',
-  function VolcanoPlotCtrl($scope, VolcanoPlotSpec) {
-    // define controller instance variables that link with vega
-    this.spec = VolcanoPlotSpec;
-    this.data = $scope.plotData;
-    // FIXME: can't get selectedNodes out this way... need to use view api?
-    // $scope.plotData.selectedNodes: $scope.selectedNodes
+// TODO when refactoring for issue #76, this service belongs in the top-level
+//      src/app/ module called `node` in its `services.js` file
+.factory('NodeService', ['$resource', 'ApiBasePath',
+  function($resource, ApiBasePath) {
+    // private methods for NodeService
+    var Participation = $resource(ApiBasePath + 'participation/');
+
+    // public methods for NodeService
+    return {
+      getGenesForNodesPromise: function getGenesForNodesPromise(nodes) {
```

Comment:
I had read somewhere that naming functions helps when you get a stack trace. I couldn't remember where I read that so I Googled around and found this blog post on the subject:
> http://www.therightcode.net/naming-functions-yes-but-not-all-the-time/
The logic there makes perfect sense: use named functions when it would help for stack traces, but don't do it in cases (such as these you've highlighted) where it doesn't actually help and causes redundancy in the code.

Will fix.

- [ ] don't use named functions on objects that already "name" them via a method